treewide: use sock_tl_name2ep() class of functions where applicable

benpicco opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md.
-->

### Contribution description

The `sock_tl_name2ep()` class of functions behave exactly as `sock_tl_str2ep()` except for when the `sock_dns` module is enabled.
In this case they will also be able to resolve a name via DNS in addition to parsing raw IP addresses.

This enables e.g. to fetch SUIT updates using a domain name instead of a hard coded IP address or supplying domain names in interactive tests.


### Testing procedure

No functional changes unless `sock_dns` is used. 


### Issues/PRs references

follow-up to #17510

fjmolinas reviewed (commented):
LGTM, could you provide some test-output where the convenience is used? (i.e. DNS enabled)

benpicco commented:
Good call!
`sock_tl_name2ep()` got confused if both an A and an AAAA record were returned but only IPv6 is enabled.
Fixed now:

#### tests/nanocoap_cli

```
2022-03-29 10:46:39,912 # > url get coap://coap.me/hello
2022-03-29 10:46:39,992 # offset 000: world

2022-03-29 10:46:42,512 # > url get coap://coap.me/large
2022-03-29 10:46:42,539 # offset 000: 
2022-03-29 10:46:42,542 #      0                   1     
2022-03-29 10:46:42,571 # offset 032:               2                 
2022-03-29 10:46:42,598 # offset 064:   3
2022-03-29 10:46:42,600 #     0 1 2 3 4 5 6 7 8 9 0 1 
2022-03-29 10:46:42,629 # offset 096: 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 
2022-03-29 10:46:42,656 # offset 128: 8 9 0 1
2022-03-29 10:46:42,658 #    +-+-+-+-+-+-+-+-+-+-+
2022-03-29 10:46:42,687 # offset 160: -+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
2022-03-29 10:46:42,715 # offset 192: -+-+-+-+-+-+
2022-03-29 10:46:42,716 #    |Ver| T |  TKL  
2022-03-29 10:46:42,745 # offset 224: |      Code     |          Messa
2022-03-29 10:46:42,773 # offset 256: ge ID           |
2022-03-29 10:46:42,774 #    +-+-+-+-+-+
2022-03-29 10:46:42,804 # offset 288: -+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
2022-03-29 10:46:42,832 # offset 320: -+-+-+-+-+-+-+-+-+-+-+
2022-03-29 10:46:42,833 #    |   To
2022-03-29 10:46:42,861 # offset 352: ken (if any, TKL bytes) ...
2022-03-29 10:46:42,862 #    +
2022-03-29 10:46:42,891 # offset 384: -+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
2022-03-29 10:46:42,920 # offset 416: -+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
2022-03-29 10:46:42,946 # offset 448: 
2022-03-29 10:46:42,949 #    |   Options (if any) ...
2022-03-29 10:46:42,949 #    
2022-03-29 10:46:42,978 # offset 480: +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-
2022-03-29 10:46:43,007 # offset 512: +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-
2022-03-29 10:46:43,033 # offset 544: +
2022-03-29 10:46:43,036 #    |1 1 1 1 1 1 1 1|    Payloa
2022-03-29 10:46:43,064 # offset 576: d (if any) ...
2022-03-29 10:46:43,065 #    +-+-+-+-+-+-+-
2022-03-29 10:46:43,094 # offset 608: +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-
2022-03-29 10:46:43,122 # offset 640: +-+-+-+-+-+-+-+-+-+
2022-03-29 10:46:43,122 # 
2022-03-29 10:46:43,123 # [...]
2022-03-29 10:46:43,123 # 
2022-03-29 10:46:43,123 #    T
2022-03-29 10:46:43,152 # offset 672: oken Length (TKL):  4-bit unsign
2022-03-29 10:46:43,181 # offset 704: ed integer.  Indicates the lengt
2022-03-29 10:46:43,208 # offset 736: h of
2022-03-29 10:46:43,210 #       the variable-length T
2022-03-29 10:46:43,239 # offset 768: oken field (0-8 bytes).  Lengths
2022-03-29 10:46:43,266 # offset 800:  9-15 are
2022-03-29 10:46:43,268 #       reserved, MUST N
2022-03-29 10:46:43,297 # offset 832: OT be sent, and MUST be processe
2022-03-29 10:46:43,325 # offset 864: d as a message
2022-03-29 10:46:43,327 #       format erro
2022-03-29 10:46:43,353 # offset 896: r.
2022-03-29 10:46:43,353 # 
2022-03-29 10:46:43,356 #    Code:  8-bit unsigned int
2022-03-29 10:46:43,385 # offset 928: eger, split into a 3-bit class (
2022-03-29 10:46:43,411 # offset 960: most
2022-03-29 10:46:43,414 #       significant bits) and
2022-03-29 10:46:43,443 # offset 992:  a 5-bit detail (least significa
2022-03-29 10:46:43,470 # offset 1024: nt bits),
2022-03-29 10:46:43,472 #       documented as c.
2022-03-29 10:46:43,501 # offset 1056: dd where c is a digit from 0 to 
2022-03-29 10:46:43,529 # offset 1088: 7 for the 3-bit
2022-03-29 10:46:43,530 #       subfield a
2022-03-29 10:46:43,559 # offset 1120: nd dd are two digits from 00 to 
2022-03-29 10:46:43,587 # offset 1152: 31 for the 5-bit
2022-03-29 10:46:43,588 #       subfield.
2022-03-29 10:46:43,617 # offset 1184:   The class can indicate a reque
2022-03-29 10:46:43,645 # offset 1216: st (0), a success
2022-03-29 10:46:43,646 #       response
2022-03-29 10:46:43,675 # offset 1248:  (2), a client error response (4
2022-03-29 10:46:43,703 # offset 1280: ), or a server error
2022-03-29 10:46:43,705 #       respo
2022-03-29 10:46:43,734 # offset 1312: nse (5).  (All other class value
2022-03-29 10:46:43,762 # offset 1344: s are reserved.)  As a
2022-03-29 10:46:43,763 #       spe
2022-03-29 10:46:43,792 # offset 1376: cial case, Code 0.00 indicates a
2022-03-29 10:46:43,821 # offset 1408: n Empty message.  In case of a
2022-03-29 10:46:43,821 #  
2022-03-29 10:46:43,850 # offset 1440:      request, the Code field ind
2022-03-29 10:46:43,879 # offset 1472: icates the Request Method; in ca
2022-03-29 10:46:43,906 # offset 1504: se of a
2022-03-29 10:46:43,908 #       response a Respons
2022-03-29 10:46:43,937 # offset 1536: e Code.  Possible values are mai
2022-03-29 10:46:43,965 # offset 1568: ntained in the
2022-03-29 10:46:43,967 #       CoAP Code R
2022-03-29 10:46:43,996 # offset 1600: egistries (Section 12.1).  The s
2022-03-29 10:46:44,024 # offset 1632: emantics of requests
2022-03-29 10:46:44,025 #       and r
2022-03-29 10:46:44,054 # offset 1664: esponses are defined in Section 
2022-03-29 10:46:44,083 # offset 1696: 5.
2022-03-29 10:46:44,083 # 
```

fjmolinas commented:
I ran the `lwip` and `lwip_sock%` tests as well, the default still passes. Trusting your test for the rest.

fjmolinas reviewed (approved):
ACK
